lib/enumerables.rb

```diff
@@ -0,0 +1,230 @@
+module Enumerable # rubocop:disable Metrics/ModuleLength
+  # my_each
+  def my_each(&block)
+    arr = self
+    arr = arr.to_a
+    count = 0
+    if arr.is_a?(Hash)
+      while count < arr.length
+        yield arr[i][0], arr[i][1]
+        count += 1
+      end
+      self
+    elsif block
+      while count < arr.length
+        yield arr[count]
+        count += 1
+      end
+      self
+    end
+    arr.to_enum
+  end
+
+  # my_each_with_index
+  def my_each_with_index(&block)
+    arr = self
+    arr = arr.to_a
+    count = 0
+    if arr.is_a?(Hash)
+      while count < arr.length
+        yield arr[i], count
+        count += 1
+      end
+      self
+    elsif block
+      while count < arr.length
+        yield arr[count], count
+        count += 1
+      end
+      self
+    else
+      arr.to_enum
+    end
+  end
+
+  # my_select
+  def my_select(&block)
+    arr = self
+    return arr.to_enum unless block
+
+    temp = []
+    arr.my_each { |num| temp.push(num) if yield(num) }
+    temp
+  end
+
+  # my_all
+  def my_all?(paramet = nil)
+    an = true
+    arr = self
+    arr = arr.to_a
+    return true if arr.empty?
+
+    if block_given?
+      arr.my_each do |n|
+        ans = yield n
+        return false if ans == false
+      end
+    else
+      an = check_regx(paramet, arr)
+    end
+    an
+  end
+
+  # my_method1
+  def check_regx(paramet, arr)
+    an = flase
+    if paramet.nil?
+      arr.my_each { |n| return false if n.nil? || !n }
+    else
```

Comment:
When no block or arguments are passed, the method should check for falsy values and return false only when the value is falsy! Your method returns false for all truth values too.

example:

```ruby
p [[], true].my_all?
```
Should return `true`. Your method returns `false`. Please fix this.